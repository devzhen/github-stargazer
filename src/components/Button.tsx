import React from 'react';
import { ActivityIndicator } from 'react-native';
import styled from 'styled-components/native';

export enum ButtonBackground {
  Orange = 'orange',
  Gray = 'gray',
}

interface ButtonProps {
  labelSize: number;
  label: string;
  bgColor: ButtonBackground;
  onPress: any;
  disabled: boolean;
  loading?: boolean;
}

const Button = ({
  labelSize,
  label,
  bgColor,
  onPress,
  disabled,
  loading,
}: ButtonProps) => (
  <Wrapper
    onPress={() => {
      if (disabled || loading) {
        return;
      }
      onPress();
    }}
    bgColor={bgColor}
    disabled={disabled}
  >
    {!loading && <Label labelSize={labelSize}>{label}</Label>}
    {loading && <ActivityIndicator size="small" color="#fff" />}
  </Wrapper>
);

Button.defaultProps = {
  onPress: () => {},
  labelSize: 17,
  label: 'No data',
  bgColor: ButtonBackground.Orange,
  disabled: false,
  loading: false,
};

const Wrapper = styled.TouchableOpacity<{ bgColor: string; disabled: boolean }>`
  flex: 1;
  height: 100%;
  align-items: center;
  flex-direction: row;
  justify-content: center;
  ${(props) =>
    props.bgColor === ButtonBackground.Orange &&
    `background-color: ${props.theme.colors.orange}`}
  ${(props) =>
    props.bgColor === ButtonBackground.Gray &&
    `background-color: ${props.theme.colors.gray}`}
  border-radius: 4px;
  ${(props) => props.disabled && 'opacity: 0.7'};
`;

const Label = styled.Text<{ labelSize: number }>`
  font-size: ${(props) => props.labelSize}px;
  text-align: center;
  color: #fff;
`;

export default Button;
