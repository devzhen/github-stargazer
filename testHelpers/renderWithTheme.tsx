import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
