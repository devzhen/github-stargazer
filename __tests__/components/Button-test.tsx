import React from 'react';
import 'jest-styled-components';

import Button from '@components/Button';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('Button common component', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Button label="test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
