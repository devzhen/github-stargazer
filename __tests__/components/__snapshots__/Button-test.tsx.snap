// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button common component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eb9e38",
      "borderRadius": 4,
      "flexBasis": 0,
      "flexDirection": "row",
      "flexGrow": 1,
      "flexShrink": 1,
      "height": "100%",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    labelSize={17}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 17,
          "textAlign": "center",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
